useragent/user_agent.go

```diff
@@ -99,13 +106,29 @@ func (u info) String() string {
 
 type data []info
 
+// Validate the key value pair being set in the user agent. Error if invalid.
+func validate(key, value string) error {
+	if !isValid(key) {
+		return fmt.Errorf("expected user agent key to be alphanumeric: %q", key)
+	}
+	if !isValid(value) {
```

Comment:
This no longer checks for semver strings, which can already be in use.

I believe the semver string is a superset of the current alphanum class, so that can be an issue.

Also, I don't think you need the `isValid` function and can avoid the additional function definition.

Reply:
done